sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -120,6 +121,19 @@ sealed trait FrameBoundary {
   def notFollows(other: FrameBoundary): Boolean
 }
 
+/**
+ * Extractor for making working with frame boundaries easier.
+ */
+object FrameBoundary {
+  def apply(boundary: FrameBoundary): Option[Int] = unapply(boundary)
+  def unapply(boundary: FrameBoundary): Option[Int] = boundary match {
+    case CurrentRow => Some(0)
+    case ValuePreceding(offset) => Some(-offset)
+    case ValueFollowing(offset) => Some(offset)
+    case _ => None
```

Comment:
I don't think there is any reason to have ```UnboundedPreceding``` and ```UnboundedFollowing``` in the first place. Their position in the frame specification should implies whether it is ```Preceding``` or ```Following```. I would like to factor this out in a followup PR.

To answer your question: I current don't care about here position, and I assume they cannot be reversed. I don't think the Hive parser allows this (I'll check later).